src/Types.jl

```diff
@@ -178,7 +178,16 @@ function projectfile_path(env_path::String; strict=false)
 end
 
 function manifestfile_path(env_path::String; strict=false)
-    for name in Base.manifest_names
+    man_names = Base.manifest_names isa Tuple ? Base.manifest_names : Base.manifest_names()
+    if first(man_names) == last(man_names) # indicative of a custom manifest name
+        man_path = joinpath(env_path, first(man_names))
+        if strict && !isfile(man_path)
+            return nothing
+        else
+            return man_path
+        end
+    end
```

Comment:
@IanButterworth I may be missing something, but now that we guarantee that `man_names` is always the same length as `Base.project_names`, can we not just delete lines 182 to 189?

Reply:
Sure but it will check for the file twice if it doesn't exist.

Reply:
Ah. Yeah, it's probably good to avoid doing unnecessary filesystem operations.